packages/progress-monitor/src/browser/progress-dialog.ts

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from "inversify";
+import { Message } from '@phosphor/messaging';
+import { Disposable } from "@theia/core";
+import { Widget, BaseWidget, Key } from "@theia/core/lib/browser";
+import { ProgressWidget } from "./progress-widget";
+
+@injectable()
+export class ProgressDialog extends BaseWidget {
+
+    protected readonly contentNode: HTMLDivElement;
+
+    protected isOpen: boolean = false;
+
+    constructor(
+        @inject(ProgressWidget) readonly widget: ProgressWidget
+    ) {
+        super();
+
+        this.contentNode = document.createElement("div");
+        this.contentNode.classList.add('theia-ProgressContent');
+
+        this.node.classList.add('theia-ProgressWidgetDialog');
```

Comment:
Please use constants. We did [this](https://github.com/theia-ide/theia/blob/c5554c8a7bfe909ca04c7bde6c95070decaf3adf/packages/git/src/browser/git-widget.tsx#L635-L648) for the Git widget for instance, but I am OK with anything that is not a magic string.